Allow to search data items by name

blizarazu opened the pull request:
Now you can find data items by shortid or name.

blizarazu committed:
Allow to search data items by name

blizarazu committed:
Allow to search data items by name

bjrmatos commented:
hi! sorry for the long wait, can you explain a little bit the use case for this change? how it makes things much easy for you? we are a little bit against to merge this, just because it will need more changes after we release [folder support](jsreport/jsreport#250), so the code here will anyway change.

blizarazu commented:
Hi, thank you for your response! I have a project were I load the templates from code giving the template's name to the render. I also have my locale files named with the language code and report name, e.g. es-template1-data. This way I can load any template with the desired language just giving the language code and template name and, without needing a record of the shortid of each locale file in my code.

```
function doRender(reportName, data, locale) {
    let header = '/path/to/header.html';
    let footer = '/path/to/footer.html';
    return jsreport.render({
      template: {
        "name": reportName,
        phantom: {
          "format": "A4",
          "headerHeight": "2cm",
          "footerHeight": "2cm",
          "header": header,
          "footer": footer
        },
        "resources": {
            "items": [
                {
                    "name": locale + "-" + reportName + "-data",
                    "entitySet": "data"
                }
             ],
            "defaultLanguage": locale
        },
      },
      data: data,
      options: {
        "language": locale,
      }
    });
  }
```

blizarazu committed:
Merge remote-tracking branch 'upstream/master'

bjrmatos committed:
remove package.json from studio folder

bjrmatos committed:
update dev deps

blizarazu committed:
Allow to search data items by name

blizarazu committed:
Change package name

bjrmatos committed:
update deps

bjrmatos committed:
update dev deps

bjrmatos committed:
fix multiple language resources and defaultLanguage

https://forum.jsreport.net/topic/1501/multiple-language-resources-and-defaultlanguage

bjrmatos committed:
update dev deps

bjrmatos committed:
release 2.0.4

bjrmatos committed:
add referenceTo metadata to templates.resources.items

pofider committed:
update dev deps

pofider committed:
release 2.1.0

bjrmatos committed:
update dev deps

blizarazu committed:
2.1.0

pofider commented:
Thank you for your work here.

We've realized that we don't like these "hard" references between template and localization data. I believe that is your main problem?
In the jsreport v3, we removed the resources extension and replaced it with a new extension that doesn't need references and better fits the folder structures. 

Could you please check the documentation of the new extension and the playground demo?
https://jsreport.net/learn/localization?version=3.0.0
https://playground.jsreport.net/w/admin/1cC6FCIe

Does it fill all your needs? It would help us a lot if you could provide us with feedback.
We should ship the final v3 release in two weeks or something close.
